modules/icgc-argo-workflows/checkinput/resources/usr/bin/rnaaln.py

```diff
@@ -0,0 +1,493 @@
+            raise AssertionError("'single_end' input is required.")
+        if row[self._single_end_col].lower()!="true" and row[self._single_end_col].lower()!="false":
+            raise AssertionError("'single_end' should be specifed as \"True\" or \"False\".")
+
+
+    def _validate_read_group_count(self, row):
+        """Assert that expected read_group_count is correct."""
+        if len(row[self._read_group_count_col]) <= 0:
+            raise AssertionError("'read_group_count' input is required.")
+
+    def _validate_experiment(self, row):
+        """Assert that expected Experiment is correct."""
+        if len(row[self._experiment_col]) <= 0:
+            raise AssertionError("'experiment' input is required.")
+        for val in ["WGS","WXS","RNA-Seq","Bisulfite-Seq","ChIP-Seq","Targeted-Seq"]:
+            if val==row[self._experiment_col]:
+                return
+        raise AssertionError("'experiment' type does not match the following: \"WGS\",\"WXS\",\"RNA-Seq\",\"Bisulfite-Seq\",\"ChIP-Seq\",\"Targeted-Seq\".")
+
+    def _validate_analysis_json(self, row):
+        """Assert that expected analysis_json is correct."""
+        if len(row[self._analysis_json_col]) <= 0:
+            raise AssertionError("'analysis_json' input is required.")
+        if not row[self._analysis_json_col].endswith(".json"):
+            raise AssertionError("'analysis_json' input should have the suffix \".json\".")
+
+    def _validate_library_name(self, row):
+        """Assert that expected library_name is correct."""
+        if len(row[self._library_name_col]) <= 0:
+            raise AssertionError("'library_name' input is required.")
+
+    def _validate_platform_unit(self, row):
+        """Assert that expected platform_unit is correct."""
+        if len(row[self._platform_unit_col]) <= 0:
+            raise AssertionError("'platform_unit' input is required.")
+
+    def _validate_platform_col(self, row):
+        """Assert that expected platform is correct."""
+        if len(row[self._platform_col]) <= 0:
+            raise AssertionError("'platform' input is required.")
+
+    def _validate_sequencing_center_col(self, row):
+        """Assert that expected sequencing_center is correct."""
+        if len(row[self._sequencing_center_col]) <= 0:
+            raise AssertionError("'sequencing_center' input is required.")
+
+    def _validate_library_strandedness_col(self, row):
+        """Assert that expected strandedness is correct"""
+        if len(row[self._library_strandedness_col]) <= 0:
+            raise AssertionError("'library_strandedness' input in required")
+        if row[self._library_strandedness_col] !="FIRST_READ_SENSE_STRAND" and row[self._library_strandedness_col] !="FIRST_READ_ANTISENSE_STRAND" and row[self._library_strandedness_col] !="UNSTRANDED":
+            raise AssertionError("library_strandedness should be one of the following values : FIRST_READ_SENSE_STRAND,FIRST_READ_ANTISENSE_STRAND,UNSTRANDED")
```

Comment:
To address Edmund's comments: added `_validate_library_strandedness_col` to validate whether the input is one of the three allowed values `FIRST_READ_SENSE_STRAND`, `FIRST_READ_ANTISENSE_STRAND` ,`UNSTRANDED` based on [Metadata Dictionary](https://docs.icgc-argo.org/docs/submission/submitting-molecular-data#metadata-dictionary) 

Later, in line 145-148:
`FIRST_READ_SENSE_STRAND` was transformed to `forward`
`FIRST_READ_ANTISENSE_STRAND` was transformed to `reverse`
`UNSTRANDED` was left as is

In HISAT2_ALIGN main.nf, all three cases (including UNSTRANDED, which is treated as null ) are taken care of by the following setting:
```
    def strandedness = meta.strandedness == 'forward' ? meta.single_end ? '--rna-strandness F' : '--rna-strandness FR' : meta.strandedness == 'reverse' ? meta.single_end ? '--rna-strandness R' : '--rna-strandness RF' : ''
```
```
    """
    INDEX=\$(find -L ${index} -name "*.1.ht2" | sed 's/\\.1.ht2\$//')
    hisat2 \\
        -t -q \\
        -x \$INDEX \\
        -p $task.cpus \\
        ${meta.single_end ? "-1 ${reads.join(",")}" : "-1 ${reads1.join(",")} -2 ${reads2.join(",")}"} \\
        $strandedness \\
        --min-intronlen 20 \\
...
```